From 871c6a0c512ecc88dbb121709cbb459bfe48f8f9 Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Wed, 26 Dec 2012 16:52:05 +0800
Subject: [PATCH 5/5] net/fsl_pq_mdio: add check for null tbipa

We don't need to set the tbi address for fman mdio.
So initialize the tbipa to null, and check for null
before writing the tbi address. This will drop the
hardcoded #ifndef CONFIG_FSL_FMAN and also fix the
null dereferencing oops when CONFIG_FSL_FMAN is not
enabled on DPAA SoCs.

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/freescale/fsl_pq_mdio.c |    7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/freescale/fsl_pq_mdio.c b/drivers/net/ethernet/freescale/fsl_pq_mdio.c
index 08daf07..fcd420a 100644
--- a/drivers/net/ethernet/freescale/fsl_pq_mdio.c
+++ b/drivers/net/ethernet/freescale/fsl_pq_mdio.c
@@ -295,7 +295,7 @@ static int fsl_pq_mdio_probe(struct platform_device *ofdev)
 	struct fsl_pq_mdio_priv *priv;
 	struct fsl_pq_mdio __iomem *regs = NULL;
 	void __iomem *map;
-	u32 __iomem *tbipa;
+	u32 __iomem *tbipa = NULL;
 	struct mii_bus *new_bus;
 	int tbiaddr = -1;
 	const u32 *addrp;
@@ -408,9 +408,8 @@ static int fsl_pq_mdio_probe(struct platform_device *ofdev)
 			err = -EBUSY;
 			goto err_free_irqs;
 		} else {
-#ifndef CONFIG_FSL_FMAN
-			out_be32(tbipa, tbiaddr);
-#endif
+			if (tbipa)
+				out_be32(tbipa, tbiaddr);
 		}
 	}
 
-- 
1.7.9.7

